### ISSUE TITLE
Type extension broken with typeof

### ISSUE THREAD

**Issue by OneCyrus** [ChilliCream/graphql-platform#6218]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Hot Chocolate

### Describe the bug

the following code to extend the query root type only works when using a string to specify the type. typeof(...) doesn't work

   ```csharp
    public class Query : ObjectType
    {
        protected override void Configure(IObjectTypeDescriptor descriptor)
        {
            descriptor.Field("person").Type<StringType>().Resolve(true);
        }
    }

    [ExtendObjectType(typeof(Query))] // doesn't work
    // [ExtendObjectType("Query")] // works
    public class QueryExtensions
    {
        public string GetExtendedMethod() => "Extended";
    }
```

### Steps to reproduce

1. extend a root type with  [ExtendObjectType(typeof(Query))] and the actual schema doesn't include this extension

### Version

13.1.0

**Comment by tobias-tengler** [comment]
I've also observed this in the past. Extensions don't work if you target the "GraphQL type". They would only work when extending a type that doesn't inherit from the GraphQL classes.
I also find this behavior weird and confusing.
Obligatory reminder though, if you use [automatic type registration](https://chillicream.com/docs/hotchocolate/v13/defining-a-schema#automatic-type-registration), you can remove all QueryType classes and registrations code and simply add the `[QueryType]` attribute to your class with extensions. All classes annotated that way will extend the query type.

**Comment by AntonC9018** [comment]
I solved this one by defining an empty class for the query, and making the `QueryType` (which is called `Query` in your situation) inherit `IObjectTypeDescriptor<Query>` (`Query` being the new empty class). Target this new `Query`.
If you happen to need `IObjectTypeDescriptor` for field definitions, just cast the descriptor you're given to it. The type that HotChocolate passes you does implement it.

**Comment by AntonC9018** [comment]
In my case though using the "Query" name constant did not do it either.

**Comment by OneCyrus** [comment]
> I've also observed this in the past. Extensions don't work if you target the "GraphQL type". They would only work when extending a type that doesn't inherit from the GraphQL classes. I also find this behavior weird and confusing. 

with 12.x it worked though. It's a regression in 13.x

**Comment by michaelstaib** [comment]
This is now fixed and will be included into 13.2.0-preview.4
